projects/driver_controls/src/direction_fsm.c

```diff
@@ -57,36 +58,39 @@ static void prv_state_neutral(FSM *fsm, const Event *e, void *context) {
   EventArbiterCheck *event_check = fsm->context;
   *event_check = prv_check_neutral;
 
-  InputEventData data;
+  EventArbiterOutputData data = {
+    .id = CAN_OUTPUT_MESSAGE_DIRECTION_SELECTOR,
+    .state = DIRECTION_FSM_STATE_NEUTRAL,
+    .data = e->data
+  };
 
-  data.components.data = e->data;
-  data.components.state = DIRECTION_FSM_STATE_NEUTRAL;
-
-  event_raise(INPUT_EVENT_CAN_ID_DIRECTION_SELECTOR, data.raw);
+  event_arbiter_output(data);
 }
 
 static void prv_state_forward(FSM *fsm, const Event *e, void *context) {
   EventArbiterCheck *event_check = fsm->context;
   *event_check = prv_check_forward;
 
-  InputEventData data;
-
-  data.components.data = e->data;
-  data.components.state = DIRECTION_FSM_STATE_FORWARD;
+  EventArbiterOutputData data = {
+    .id = CAN_OUTPUT_MESSAGE_DIRECTION_SELECTOR,
+    .state = DIRECTION_FSM_STATE_FORWARD,
+    .data = e->data
```

Comment:
Probably add comments everywhere you're using event data

Reply:
Many of the digital events don't use the data field, so it might be clearer to pass in 0 in those cases.

Reply:
👍